tests/main/remodel/task.yaml

```diff
@@ -66,17 +75,16 @@ execute: |
     echo "and we got the new required snap"
     snap list test-snapd-tools
 
-    # FIXME: check that this is the active model assertion
     echo "and we got the new model assertion"
     snap debug model|MATCH "revision: 2"
 
-    # FIXME: this fails (meaning, the remove does not fail)
-    # echo "We can't remove a required snap"
-    # ! snap remove test-snapd-tools
+    echo "and we cannot remove the new required snap"
+    ! snap remove test-snapd-tools
 
-    # when the above no longer fails, this will fail, so you'll have to
-    # re-remodel to be able to remove the snap (or clean up manually in restore
-    # -- but better to check here for the expected required -> non-required
-    # behaviour):
-    echo "Remove the ex-required snap"
+    echo "And we can remodel again this time test-snapd-tools is no longer required"
+    snap remodel "$TESTSLIB"/assertions/developer1-pc-revno3.model
+    snap debug model|MATCH "revision: 3"
+    echo "and test-snapd-tools is still available"
+    snap list test-snapd-tools
+    echo "and we can clean it up here because it is no longer required"
```

Comment:
Could you also add a test that shows "snap changes" for the remodel change and shows what happened there, roughly? A list of tasks would be nice.

Reply:
Added that now, thank you!